memory_engine/memory_store.py

```
@@ -293,8 +294,11 @@ def load(self) -> bool:
                     )
 
                 if os.path.exists(resolved_path):
-                    with open(resolved_path, "rb") as f:
-                        self.memory = pickle.load(f)  # nosec B301 - trusted internal persistence format
+                    # Use inter-process file lock to prevent concurrent access corruption
+                    lock_path = resolved_path + ".lock"
+                    with fasteners.InterProcessLock(lock_path):
+                        with open(resolved_path, "rb") as f:
+                            self.memory = pickle.load(f)  # nosec B301 - trusted internal persistence format
```

Ruff (0.14.10)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)